src/ExtInterface/ReefMod/MatDomain.jl

```diff
@@ -0,0 +1,276 @@
+using ADRIA: SimConstants, Domain, GDF
+using AxisKeys
+using DataFrames
+using DimensionalData
+using Distributions
+using MAT
+using ModelParameters
+using NamedDims
+using NetCDF
+using Statistics
+using YAXArrays
+
+using Infiltrator
+
+
+mutable struct ReefModDomain <: Domain
+    const name::String
+    RCP::String
+    env_layer_md
+    scenario_invoke_time::String  # time latest set of scenarios were run
+    const conn
+    const in_conn
+    const out_conn
+    const strong_pred
+    const site_data
+    const site_id_col
+    const cluster_id_col
+    init_coral_cover
+    const coral_growth::CoralGrowth
+    const site_ids
+    dhw_scens
+
+    # `wave_scens` holds empty wave data to maintain compatibility with
+    # ADRIA's dMCDA methods
+    wave_scens
+
+    # `cyclone_mortality_scens` holds dummy cyclone mortality data to maintain compatibility
+    # with ADRIA's dMCDA methods
+    cyclone_mortality_scens
+
+    model
+    sim_constants::SimConstants
+    netcdf_dir::String
+    loc_ids_path::String
+end
+
+"""
+    load_domain(
+        ::Type{ReefModDomain},
+        netcdf_dir::String,
+        geo_path::String,
+        RCP::String
+    )::ReefModDomain
+
+Load ReefMod Matlab Dataset stored in netcdf file format. 
+Uses a path ReefMod Engine data to fill missing required data
+
+# Arguments
+- `ReefModDomain` : DataType
+- `netcdf_dir` : path to netcdf ReefMod Matlab Dataset Directory
+- `geo_path` : path to geo data used in RMEDomain
+
+# Returns
+RMEDomain
+"""
+function load_domain(
+    ::Type{ReefModDomain},
+    netcdf_dir::String,
+    rfmod_engine_path::String,
+    RCP::String,
+    timeframe =  (2022, 2100)
+)::ReefModDomain
+    netcdf_file = _find_netcdf(netcdf_dir, RCP)
+    dom_dataset::Dataset = open_dataset(netcdf_file)
+    data_files = joinpath(rfmod_engine_path, "data_files")
+    loc_ids_path = joinpath(data_files, "dhw")
+    loc_ids = _get_loc_ids(loc_ids_path)
```

Comment:
What's loc_ids_path for in this instance?

Shouldn't we be extracting these based on the data inside the MAT files? There's no guarentee RME and MAT datasets will align at all.

Reply:
Oh I see what it's being used for.

In any case, the location ids should be extracted from the geospatial data, not DHW data